pkg/registry/storage/csidriver/strategy.go

```diff
@@ -50,6 +50,9 @@ func (csiDriverStrategy) PrepareForCreate(ctx context.Context, obj runtime.Objec
 	if !utilfeature.DefaultFeatureGate.Enabled(features.SELinuxMountReadWriteOncePod) {
 		csiDriver.Spec.SELinuxMount = nil
 	}
+	if !utilfeature.DefaultFeatureGate.Enabled(features.MutableCSINodeAllocatableCount) {
```

Comment:
We need something similar for ErrorCode

Reply:
Got it, I've updated the strategy to clear the field in `PrepareForUpdate`. 

I believe it shouldn't be necessary to clear ErrorCode in the create path since the entire status field is already being cleared:

https://github.com/kubernetes/kubernetes/blob/22ff6b4918e9cec09842d6ca94584b81a03042cd/pkg/registry/storage/volumeattachment/strategy.go#L59